Automattic/wp-calypso
# Jetpack SSO: Convert SSO to use path arguments instead of query params

After requesting the `/jetpack/sso/?` route be whitelisted, Barry suggested that we convert Jetpack SSO to use path arguments instead of query parameters.

This PR makes that happen.

See D1921 for related WPCOM changes.

To test:
- Checkout `update/jetpack-sso-routing` branch
- Run tests: `npm run test-client -- --grep="state jetpack-connect"`
- Checkout D1921 in WPCOM sandbox
- Check out latest master in Jetpack site
- Click "Log in with WordPress.com" button on remote site
- You will be redirect to `wpcalypso.wordpress.com/jetpack/sso/$site_id/$sso_nonce`
- Swap out `calypso.localhost:3000` for `wpcalypso.wordpress.com`
- Does it work?

cc @lezama for testing
